script/tool/test/common/package_looping_command_test.dart

```diff
@@ -185,6 +185,23 @@ void main() {
             package.childDirectory('example').path,
           ]));
     });
+
+    test('excludes subpackages when main package is excluded', () async {
+      createFakePlugin('a_plugin', packagesDir,
```

Comment:
IMO this would be slightly more clear if you retained a reference to `a_plugin` and asserted that it didn't show up in the output.  Not capturing the output to `createFakePlugin` makes it a bit less clear where the side effects are.

Reply:
Done.